src/foam/core/Promised.js

```diff
@@ -129,38 +109,35 @@ foam.CLASS({
       var myAxioms = [
         foam.core.Proxy.create({
           name: stateName,
-          flags: ['js', 'swift'],
           of: this.of,
-          delegates: methodNames,
-          forwards: [],
+          forwards: methodNames,
           factory: function() {
-            return this[pendingState].create();
+            return this[pendingState].create({obj: this});
```

Comment:
spacing should be `.create({ obj: this });`

Reply:
Done